paper-api/src/main/java/io/papermc/paper/event/player/PlayerMapFilledEvent.java

```diff
@@ -0,0 +1,58 @@
+package io.papermc.paper.event.player;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jspecify.annotations.NullMarked;
+
+/**
+ * Called when a player creates a filled map by right-clicking an empty map.
+ */
+@NullMarked
+public class PlayerMapFilledEvent extends PlayerEvent {
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+
+    private final ItemStack originalItem;
+    private ItemStack createdMap;
+
+    @ApiStatus.Internal
+    public PlayerMapFilledEvent(final Player player, final ItemStack originalItem, final ItemStack createdMap) {
+        super(player);
+        this.originalItem = originalItem;
+        this.createdMap = createdMap;
+    }
+
+    /**
+     * Returns a copy of the empty map before it was consumed.
+     */
+    public ItemStack getOriginalItem() {
+        return this.originalItem.clone();
+    }
+
+    /**
+     * Returns the filled map which was created.
```

Comment:
above method javadocs say "copy of" and this one does not, despite also being a copy